sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs

```diff
@@ -60,5 +60,76 @@ public void SetProperty(string name, object value)
 
             _properties[name] = value;
         }
+
+        public Stream? PreserveResponseContent()
+        {
+            switch (_response?.ContentStream)
+            {
+                case ResponseShouldNotBeUsedStream responseContent:
+                    return responseContent.Original;
+                case Stream stream :
+                    _response.ContentStream = new ResponseShouldNotBeUsedStream(_response.ContentStream);
+                    return stream;
+                default:
+                    return null;
+            }
+        }
+
+        public void Dispose()
+        {
+            Request?.Dispose();
+            _response?.Dispose();
+        }
+
+        private class ResponseShouldNotBeUsedStream: Stream
+        {
+            public Stream Original { get; }
+
+            public ResponseShouldNotBeUsedStream(Stream original)
+            {
+                Original = original;
+            }
+
+            private static Exception CreateException()
+            {
+                return new InvalidOperationException("This operation returns Stream as part of the model type. It should be used instead of the response content stream.");
```

Comment:
I worry about users understanding this error message because "model type" isn't as obvious to new users.  Maybe we say this operation has called PreserveResponseContent and will provide the stream as part of its response type?  Then they can quickly search for PreserveResponseContent in the docs and we have a chance to make them read a few sentences and see some examples.